ListingManager/Program.cs

```diff
@@ -2,84 +2,67 @@
 
 namespace EssentialCSharp.ListingManager;
 
-public class Program
+public sealed class Program
 {
-    private const string IntelliTect =
-@" _____       _       _ _    _______        _   
-  |_   _|     | |     | | ( )|__   __|      | |  
-    | |  _ __ | |_ ___| | |_    | | ___  ___| |_ 
-    | | | '_ \| __/ _ \ | | |   | |/ _ \/ __| __|
-   _| |_| | | | ||  __/ | | |   | |  __/ (__| |_ 
-  |_____|_| |_|\__\___|_|_|_|   |_|\___|\___|\__|";
-    private static int Main(string[] args)
+    private static Task<int> Main(string[] args)
     {
-        var directoryIn = new Option<DirectoryInfo>(
-            name: "--path",
-            description: "The directory of the chapter to update listings on.")
-        { IsRequired = true };
-
-        // With proper logging implemented, this option will hopefully be removed
-        var verboseOption = new Option<bool>(
-            name: "--verbose",
-            description: "Displays more detailed messages in the log");
-
-        var previewOption = new Option<bool>(
-            name: "--preview",
-            description: "Writes all logs to console as if changes will be made without actually making changes.");
-
-        var allChaptersOption = new Option<bool>(
-           name: "--allChapters",
-           description: "The passed in path is the parent directory to many chapter directories rather than a single chapter directory.");
-
-        // TODO: Add better descriptions when their functionality becomes clearer
-        var byFolderOption = new Option<bool>(
-            name: "--byfolder",
-            description: "");
+        CliConfiguration configuration = GetConfiguration();
+        return configuration.InvokeAsync(args);
+    }
 
-        var singleDirOption = new Option<bool>(
-            name: "--singleDir",
-            description: "All listings are in a single directory and not separated into chapter and chapter test directories");
+    public static CliConfiguration GetConfiguration()
+    {
+        CliArgument<DirectoryInfo> directoryInArgument = new("path")
```

Comment:
There is an `ExistingOnly` method you can call on this to ensure that only existing directories are valid.

Reply:
Like this? https://github.com/IntelliTect/EssentialCSharp.ListingManager/pull/110/commits/d35c1ec17b49d012a3c47bad6e50c36353d9a1ee

Reply:
yep